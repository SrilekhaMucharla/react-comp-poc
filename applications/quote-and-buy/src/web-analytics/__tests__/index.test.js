import React from 'react';
import { shallow } from 'enzyme';
import {
    HDButtonRefactor,
    HDButtonDashed,
    HDToggleButtonGroupRefactor,
    HDDropdownList,
    HDTextInputRefactor,
    HDAsyncSelect,
    HDDatePickerRefactor,
    HDSwitch,
    HDTable,
    HDQuoteDownloadRefactor,
    HDLabelRefactor,
    HDCheckboxButtonList,
    HDRadioButtonList,
    HDCheckbox,
    HDCompletedCardInfo
} from 'hastings-components';
import {
    AnalyticsHDButton,
    AnalyticsHDToggleButtonGroup,
    AnalyticsHDDropdownList,
    AnalyticsHDTextInput,
    AnalyticsHDOverlayPopup,
    AnalyticsHDAsyncSelect,
    AnalyticsHDButtonDashed,
    AnalyticsHDModal,
    AnalyticsHDDatePicker,
    AnalyticsHDSwitch,
    AnalyticsHDTable,
    AnalyticsHDQuoteDownload,
    AnalyticsHDLabel,
    AnalyticsHDCheckboxButtonList,
    AnalyticsHDRadioButtonList,
    AnalyticsHDCheckbox,
    AnalyticsHDCompletedCardInfo
} from '../index';


describe('Elements with analytics', () => {
    // given
    const webAnalyticsEvent = {
        analytics_data: 'top secret'
    };
    const otherProps = {
        otherProp: 'some value'
    };
    const id = 'default-id';
    const arrayValues = [{
        value: 'value 1',
        name: 'name 1'
    }, {
        value: 'value 2',
        name: 'name 2'
    }];


    // helpers
    const shouldHaveFunctionsDefined = (wrapper, functions) => {
        const possibleFunctions = [
            'onChange', 'onClick', 'onFocus', 'onBlur', 'onConfirm', 'onCancel', 'onBeforeOpen', 'onSelect'
        ];

        possibleFunctions.forEach((funcName) => {
            if (functions.includes(funcName)) {
                expect(wrapper.props()[funcName]).toBeDefined();
            } else {
                expect(wrapper.props()[funcName] === undefined || wrapper.props()[funcName] === null).toBeTruthy();
            }
        });
    };

    const shouldPassOtherPropsUnchanged = (wrapper) => {
        expect(wrapper.props().otherProp).toBe(otherProps.otherProp);
    };

    describe('AnalyticsHDButtonDashed', () => {
        const getWrapper = () => shallow(<AnalyticsHDButtonDashed webAnalyticsEvent={webAnalyticsEvent} id={id} {...otherProps} />);
        it('should render and match the snapshot', () => {
            // given
            const wrapper = getWrapper();
            // then
            expect(wrapper).toMatchSnapshot();
        });

        it('should pass functions to component when webAnalyticsEvent prop is provided', () => {
            // given
            const wrapper = getWrapper();
            // then
            shouldHaveFunctionsDefined(wrapper, ['onClick']);
            shouldPassOtherPropsUnchanged(wrapper);
        });

        it('should return the same component if webAnalyticsEvent is not provided', () => {
            // given
            const wrapper = shallow(<AnalyticsHDButtonDashed id={id} {...otherProps} />);
            // then
            expect(wrapper.type()).toBe(HDButtonDashed);
            shouldHaveFunctionsDefined(wrapper, ['onClick']);
            shouldPassOtherPropsUnchanged(wrapper);
        });
    });

    describe('AnalyticsHDModal', () => {
        const getWrapper = () => shallow(
            <AnalyticsHDModal
                show
                onConfirm={() => {}}
                webAnalyticsEvent={webAnalyticsEvent}
                id={id}
                {...otherProps}
            >
                <div>Some children</div>
            </AnalyticsHDModal>
        );
        it('should render and match the snapshot', () => {
            // given
            const wrapper = getWrapper();
            // then
            expect(wrapper).toMatchSnapshot();
        });

        it('should pass functions to component when webAnalyticsEvent prop is provided', () => {
            // given
            const wrapper = getWrapper();
            // then
            shouldHaveFunctionsDefined(wrapper, ['onConfirm', 'onCancel']);
            shouldPassOtherPropsUnchanged(wrapper);
        });

        it('should return the same component if webAnalyticsEvent is not provided', () => {
            // given
            const wrapper = shallow(<AnalyticsHDModal show onConfirm={() => {}} id={id} {...otherProps}><div>Some children</div></AnalyticsHDModal>);
            // then
            shouldHaveFunctionsDefined(wrapper, ['onConfirm', 'onCancel']);
            shouldPassOtherPropsUnchanged(wrapper);
        });
    });


    describe('AnalyticsHDButton', () => {
        const getWrapper = () => shallow(<AnalyticsHDButton webAnalyticsEvent={webAnalyticsEvent} id={id} {...otherProps} />);
        it('should render and match the snapshot', () => {
            // given
            const wrapper = getWrapper();
            // then
            expect(wrapper).toMatchSnapshot();
        });

        it('should pass functions to component when webAnalyticsEvent prop is provided', () => {
            // given
            const wrapper = getWrapper();
            // then
            shouldHaveFunctionsDefined(wrapper, ['onClick']);
            shouldPassOtherPropsUnchanged(wrapper);
        });

        it('should return the same component if webAnalyticsEvent is not provided', () => {
            // given
            const wrapper = shallow(<AnalyticsHDButton id={id} {...otherProps} />);
            // then
            expect(wrapper.type()).toBe(HDButtonRefactor);
            shouldHaveFunctionsDefined(wrapper, ['onClick']);
            shouldPassOtherPropsUnchanged(wrapper);
        });
    });

    describe('AnalyticsHDToggleButtonGroup', () => {
        const getWrapper = () => shallow(
            <AnalyticsHDToggleButtonGroup availableValues={arrayValues} webAnalyticsEvent={webAnalyticsEvent} id={id} {...otherProps} />
        );
        it('should render and match the snapshot', () => {
            // given
            const wrapper = getWrapper();
            // then
            expect(wrapper).toMatchSnapshot();
        });

        it('should pass functions to component when webAnalyticsEvent prop is provided', () => {
            // given
            const wrapper = getWrapper();
            // then
            shouldHaveFunctionsDefined(wrapper, ['onChange']);
            shouldPassOtherPropsUnchanged(wrapper);
        });

        it('should return the same component if webAnalyticsEvent is not provided', () => {
            // given
            const wrapper = shallow(<AnalyticsHDToggleButtonGroup availableValues={arrayValues} id={id} {...otherProps} />);
            // then
            expect(wrapper.type()).toBe(HDToggleButtonGroupRefactor);
            shouldHaveFunctionsDefined(wrapper, ['onChange']);
            shouldPassOtherPropsUnchanged(wrapper);
        });
    });

    describe('AnalyticsHDDropdownList', () => {
        const getWrapper = () => shallow(<AnalyticsHDDropdownList options={arrayValues} webAnalyticsEvent={webAnalyticsEvent} id={id} {...otherProps} />);
        it('should render and match the snapshot', () => {
            // given
            const wrapper = getWrapper();
            // then
            expect(wrapper).toMatchSnapshot();
        });

        it('should pass functions to component when webAnalyticsEvent prop is provided', () => {
            // given
            const wrapper = getWrapper();
            // then
            shouldHaveFunctionsDefined(wrapper, ['onChange', 'onBlur']);
            shouldPassOtherPropsUnchanged(wrapper);
        });

        it('should return the same component if webAnalyticsEvent is not provided', () => {
            // given
            const wrapper = shallow(<AnalyticsHDDropdownList options={arrayValues} id={id} {...otherProps} />);
            // then
            expect(wrapper.type()).toBe(HDDropdownList);
            shouldHaveFunctionsDefined(wrapper, ['onChange', 'onBlur']);
            shouldPassOtherPropsUnchanged(wrapper);
        });
    });

    describe('AnalyticsHDTextInput', () => {
        const getWrapper = () => shallow(<AnalyticsHDTextInput webAnalyticsEvent={webAnalyticsEvent} id={id} {...otherProps} />);
        it('should render and match the snapshot', () => {
            // given
            const wrapper = getWrapper();
            // then
            expect(wrapper).toMatchSnapshot();
        });

        it('should pass functions to component when webAnalyticsEvent prop is provided', () => {
            // given
            const wrapper = getWrapper();
            // then
            shouldHaveFunctionsDefined(wrapper, ['onChange', 'onBlur', 'onFocus']);
            shouldPassOtherPropsUnchanged(wrapper);
        });

        it('should return the same component if webAnalyticsEvent is not provided', () => {
            // given
            const wrapper = shallow(<AnalyticsHDTextInput id={id} {...otherProps} />);
            // then
            expect(wrapper.type()).toBe(HDTextInputRefactor);
            shouldHaveFunctionsDefined(wrapper, ['onChange', 'onBlur', 'onFocus']);
            shouldPassOtherPropsUnchanged(wrapper);
        });
    });

    describe('AnalyticsHDOverlayPopup', () => {
        const getWrapper = () => shallow(
            <AnalyticsHDOverlayPopup
                webAnalyticsEvent={webAnalyticsEvent}
                id={id}
                {...otherProps}
            >
                <div>Some children</div>
            </AnalyticsHDOverlayPopup>
        );
        it('should render and match the snapshot', () => {
            // given
            const wrapper = getWrapper();
            // then
            expect(wrapper).toMatchSnapshot();
        });

        it('should pass functions to component when webAnalyticsEvent prop is provided', () => {
            // given
            const wrapper = getWrapper();
            // then
            shouldHaveFunctionsDefined(wrapper, ['onConfirm', 'onCancel']);
            shouldPassOtherPropsUnchanged(wrapper);
        });

        it('should return the same component if webAnalyticsEvent is not provided', () => {
            // given
            const wrapper = shallow(<AnalyticsHDOverlayPopup id={id} {...otherProps}><div>Some children</div></AnalyticsHDOverlayPopup>);
            // then
            shouldHaveFunctionsDefined(wrapper, ['onConfirm', 'onCancel']);
            shouldPassOtherPropsUnchanged(wrapper);
        });
    });

    describe('AnalyticsHDAsyncSelect', () => {
        const getWrapper = () => shallow(<AnalyticsHDAsyncSelect webAnalyticsEvent={webAnalyticsEvent} id={id} {...otherProps} />);
        it('should render and match the snapshot', () => {
            // given
            const wrapper = getWrapper();
            // then
            expect(wrapper).toMatchSnapshot();
        });

        it('should pass functions to component when webAnalyticsEvent prop is provided', () => {
            // given
            const wrapper = getWrapper();
            // then
            shouldHaveFunctionsDefined(wrapper, ['onChange', 'onBlur']);
            shouldPassOtherPropsUnchanged(wrapper);
        });

        it('should return the same component if webAnalyticsEvent is not provided', () => {
            // given
            const wrapper = shallow(<AnalyticsHDAsyncSelect id={id} {...otherProps} />);
            // then
            expect(wrapper.type()).toBe(HDAsyncSelect);
            shouldHaveFunctionsDefined(wrapper, ['onChange', 'onBlur']);
            shouldPassOtherPropsUnchanged(wrapper);
        });
    });

    describe('AnalyticsHDDatePicker', () => {
        const getWrapper = () => shallow(<AnalyticsHDDatePicker
            webAnalyticsEvent={webAnalyticsEvent}
            id={id}
            {...otherProps} />);
        it('should render and match the snapshot', () => {
            // given
            const wrapper = getWrapper();
            // then
            expect(wrapper).toMatchSnapshot();
        });

        it('should pass functions to component when webAnalyticsEvent prop is provided', () => {
            // given
            const wrapper = getWrapper();
            // then
            shouldHaveFunctionsDefined(wrapper, ['onChange', 'onBlur', 'onConfirm', 'onSelect']);
            shouldPassOtherPropsUnchanged(wrapper);
        });

        it('should return the same component if webAnalyticsEvent is not provided', () => {
            // given
            const wrapper = shallow(<AnalyticsHDDatePicker id={id} {...otherProps} />);
            // then
            expect(wrapper.type()).toBe(HDDatePickerRefactor);
            shouldHaveFunctionsDefined(wrapper, ['onChange', 'onBlur', 'onConfirm', 'onSelect']);
            shouldPassOtherPropsUnchanged(wrapper);
        });
    });

    describe('AnalyticsHDSwitch', () => {
        const getWrapper = () => shallow(<AnalyticsHDSwitch
            values={arrayValues}
            value={arrayValues[0].name}
            path="path"
            name="name"
            webAnalyticsEvent={webAnalyticsEvent}
            id={id}
            {...otherProps} />);
        it('should render and match the snapshot', () => {
            // given
            const wrapper = getWrapper();
            // then
            expect(wrapper).toMatchSnapshot();
        });

        it('should pass functions to component when webAnalyticsEvent prop is provided', () => {
            // given
            const wrapper = getWrapper();
            // then
            shouldHaveFunctionsDefined(wrapper, ['onChange']);
            shouldPassOtherPropsUnchanged(wrapper);
        });

        it('should return the same component if webAnalyticsEvent is not provided', () => {
            // given
            const wrapper = shallow(<AnalyticsHDSwitch
                values={arrayValues}
                value={arrayValues[0].name}
                path="path"
                name="name"
                id={id}
                {...otherProps} />);
            // then
            expect(wrapper.type()).toBe(HDSwitch);
            shouldHaveFunctionsDefined(wrapper, ['onChange']);
            shouldPassOtherPropsUnchanged(wrapper);
        });
    });


    describe('AnalyticsHDTable', () => {
        const getWrapper = () => shallow(<AnalyticsHDTable
            headerValues={[
                {
                    value: 'value',
                    image: <div>img</div>,
                    headerLabelValue: 'headerLabelValue',
                    secondaryLabel: 'secondaryLabel',
                    stickyHeaderText: 'stickyHeaderText'
                }
            ]}
            name="name"
            onSelect={() => {}}
            webAnalyticsEvent={webAnalyticsEvent}
            id={id}
            {...otherProps} />);
        it('should render and match the snapshot', () => {
            // given
            const wrapper = getWrapper();
            // then
            expect(wrapper).toMatchSnapshot();
        });

        it('should pass functions to component when webAnalyticsEvent prop is provided', () => {
            // given
            const wrapper = getWrapper();
            // then
            shouldHaveFunctionsDefined(wrapper, ['onSelect']);
            shouldPassOtherPropsUnchanged(wrapper);
        });

        it('should return the same component if webAnalyticsEvent is not provided', () => {
            // given
            const wrapper = shallow(<AnalyticsHDTable
                headerValues={[
                    {
                        value: 'value',
                        image: <div>img</div>,
                        headerLabelValue: 'headerLabelValue',
                        secondaryLabel: 'secondaryLabel',
                        stickyHeaderText: 'stickyHeaderText'
                    }
                ]}
                name="name"
                onSelect={() => {}}
                id={id}
                {...otherProps} />);
            // then
            expect(wrapper.type()).toBe(HDTable);
            shouldHaveFunctionsDefined(wrapper, ['onSelect']);
            shouldPassOtherPropsUnchanged(wrapper);
        });
    });


    describe('AnalyticsHDQuoteDownload', () => {
        const getWrapper = () => shallow(<AnalyticsHDQuoteDownload
            linkText="Link"
            showIcon={false}
            onClick={() => {}}
            webAnalyticsEvent={webAnalyticsEvent}
            id={id}
            {...otherProps} />);
        it('should render and match the snapshot', () => {
            // given
            const wrapper = getWrapper();
            // then
            expect(wrapper).toMatchSnapshot();
        });

        it('should pass functions to component when webAnalyticsEvent prop is provided', () => {
            // given
            const wrapper = getWrapper();
            // then
            shouldHaveFunctionsDefined(wrapper, ['onClick']);
            shouldPassOtherPropsUnchanged(wrapper);
        });

        it('should return the same component if webAnalyticsEvent is not provided', () => {
            // given
            const wrapper = shallow(<AnalyticsHDQuoteDownload
                linkText="Link"
                showIcon={false}
                onClick={() => {}}
                id={id}
                {...otherProps} />);
            // then
            expect(wrapper.type()).toBe(HDQuoteDownloadRefactor);
            shouldHaveFunctionsDefined(wrapper, ['onClick']);
            shouldPassOtherPropsUnchanged(wrapper);
        });
    });

    describe('AnalyticsHDLabel', () => {
        const getWrapper = () => shallow(<AnalyticsHDLabel
            Tag="r"
            text="Something"
            className="label-with-icon align-items-start"
            id={id}
            iconPosition="l"
            adjustImagePosition={false}
            href="https://www.hastingsdirect.com"
            target="SomeTarget"
            webAnalyticsEvent={webAnalyticsEvent}
            {...otherProps} />);
        it('should render and match the snapshot', () => {
            // given
            const wrapper = getWrapper();
            // then
            expect(wrapper).toMatchSnapshot();
        });

        it('should pass functions to component when webAnalyticsEvent prop is provided', () => {
            // given
            const wrapper = getWrapper();
            // then
            shouldPassOtherPropsUnchanged(wrapper);
        });

        it('should return the same component if webAnalyticsEvent is not provided', () => {
            // given
            const wrapper = shallow(<AnalyticsHDLabel
                Tag="r"
                text="Something"
                className="label-with-icon align-items-start"
                id={id}
                iconPosition="l"
                adjustImagePosition={false}
                href="https://www.hastingsdirect.com"
                target="SomeTarget"
                {...otherProps} />);
            // then
            expect(wrapper.type()).toBe(HDLabelRefactor);
            shouldPassOtherPropsUnchanged(wrapper);
        });
    });

    describe('AnalyticsHDCheckboxButtonList', () => {
        const getWrapper = () => shallow(<AnalyticsHDCheckboxButtonList
            className="Something"
            colProps={{ xs: 12, lg: 10, xl: 8 }}
            items={[]}
            selectedItems={[]}
            onChange={() => {}}
            webAnalyticsEvent={webAnalyticsEvent} />);
        it('should render and match the snapshot', () => {
            // given
            const wrapper = getWrapper();
            // then
            expect(wrapper).toMatchSnapshot();
        });

        it('should pass functions to component when webAnalyticsEvent prop is provided', () => {
            // given
            const wrapper = getWrapper();
            // then
            shouldHaveFunctionsDefined(wrapper, ['onChange']);
        });

        it('should return the same component if webAnalyticsEvent is not provided', () => {
            // given
            const wrapper = shallow(<AnalyticsHDCheckboxButtonList
                className="Something"
                colProps={{ xs: 12, lg: 10, xl: 8 }}
                items={[]}
                selectedItems={[]}
                onChange={() => {}} />);
            // then
            expect(wrapper.type()).toBe(HDCheckboxButtonList);
            shouldHaveFunctionsDefined(wrapper, ['onChange']);
        });
    });

    describe('AnalyticsHDRadioButtonList', () => {
        const innerValue = {
            id: 'SomeID',
            label: 'Some Label',
            value: 'Required Value'
        };
        const getWrapper = () => shallow(<AnalyticsHDRadioButtonList
            className="Something"
            items={[]}
            value={innerValue}
            onChange={() => {}}
            webAnalyticsEvent={webAnalyticsEvent} />);
        it('should render and match the snapshot', () => {
            // given
            const wrapper = getWrapper();
            // then
            expect(wrapper).toMatchSnapshot();
        });

        it('should pass functions to component when webAnalyticsEvent prop is provided', () => {
            // given
            const wrapper = getWrapper();
            // then
            shouldHaveFunctionsDefined(wrapper, ['onChange']);
        });

        it('should return the same component if webAnalyticsEvent is not provided', () => {
            // given
            const wrapper = shallow(<AnalyticsHDRadioButtonList
                className="Something"
                items={[]}
                value={innerValue}
                onChange={() => {}} />);
            // then
            expect(wrapper.type()).toBe(HDRadioButtonList);
            shouldHaveFunctionsDefined(wrapper, ['onChange']);
        });
    });

    describe('AnalyticsHDCheckbox', () => {
        const getWrapper = () => shallow(<AnalyticsHDCheckbox
            path="path.to.toggle-button"
            name="toggle-button-name"
            value="Some value"
            label="Some Label"
            className="some-class-name"
            doReset={false}
            data="String"
            type="checkbox"
            text="Sample text"
            onChange={() => {}}
            webAnalyticsEvent={webAnalyticsEvent} />);
        it('should render and match the snapshot', () => {
            // given
            const wrapper = getWrapper();
            // then
            expect(wrapper).toMatchSnapshot();
        });

        it('should pass functions to component when webAnalyticsEvent prop is provided', () => {
            // given
            const wrapper = getWrapper();
            // then
            shouldHaveFunctionsDefined(wrapper, ['onChange']);
        });

        it('should return the same component if webAnalyticsEvent is not provided', () => {
            // given
            const wrapper = shallow(<AnalyticsHDCheckbox
                path="path.to.toggle-button"
                name="toggle-button-name"
                value="Some value"
                label="Some Label"
                className="some-class-name"
                doReset={false}
                data="String"
                type="checkbox"
                text="sample text"
                onChange={() => {}} />);
            // then
            expect(wrapper.type()).toBe(HDCheckbox);
            shouldHaveFunctionsDefined(wrapper, ['onChange']);
        });
    });

    describe('AnalyticsHDCompletedCardInfo', () => {
        const getWrapper = () => shallow(<AnalyticsHDCompletedCardInfo
            text="Some sample text"
            additionalText="Some additional text"
            variant="driver"
            editTabIndex={0}
            deleteTabIndex={1}
            onEdit={() => {}}
            onDelete={() => {}}
            onEditKeyDown={() => {}}
            onDeleteKeyDown={() => {}}
            webAnalyticsEvent={webAnalyticsEvent} />);
        it('should render and match the snapshot', () => {
            // given
            const wrapper = getWrapper();
            // then
            expect(wrapper).toMatchSnapshot();
        });

        it('should pass functions to component when webAnalyticsEvent prop is provided', () => {
            // given
            const wrapper = getWrapper();
            // then
            shouldHaveFunctionsDefined(wrapper, ['onEdit']);
            shouldHaveFunctionsDefined(wrapper, ['onDelete']);
            shouldHaveFunctionsDefined(wrapper, ['onEditKeyDown']);
            shouldHaveFunctionsDefined(wrapper, ['onDeleteKeyDown']);
        });

        it('should return the same component if webAnalyticsEvent is not provided', () => {
            // given
            const wrapper = shallow(<AnalyticsHDCompletedCardInfo
                text="Some sample text"
                additionalText="Some additional text"
                variant="driver"
                editTabIndex={0}
                deleteTabIndex={1}
                onEdit={() => {}}
                onDelete={() => {}}
                onEditKeyDown={() => {}}
                onDeleteKeyDown={() => {}} />);
            // then
            expect(wrapper.type()).toBe(HDCompletedCardInfo);
            shouldHaveFunctionsDefined(wrapper, ['onEdit']);
            shouldHaveFunctionsDefined(wrapper, ['onDelete']);
            shouldHaveFunctionsDefined(wrapper, ['onEditKeyDown']);
            shouldHaveFunctionsDefined(wrapper, ['onDeleteKeyDown']);
        });
    });
});
