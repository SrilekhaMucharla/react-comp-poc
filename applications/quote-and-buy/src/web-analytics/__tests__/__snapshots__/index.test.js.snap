// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elements with analytics AnalyticsHDAsyncSelect should render and match the snapshot 1`] = `
<HDAsyncSelect
  cacheOptions={false}
  className=""
  customClassName=""
  defaultOptions={false}
  defaultValue={null}
  displayValidationMessage={true}
  id="default-id"
  label={null}
  loadOptions={[Function]}
  name="dropdown-name"
  onBlur={[Function]}
  onChange={[Function]}
  onInputChange={[Function]}
  options={null}
  otherProp="some value"
  path="path.to.dropdown"
  placeholder="text"
  selectSize={null}
  value={null}
/>
`;

exports[`Elements with analytics AnalyticsHDButton should render and match the snapshot 1`] = `
<HDButtonRefactor
  className=""
  disabled={false}
  fullWidth={false}
  id="default-id"
  label="button"
  onClick={[Function]}
  onKeyPress={null}
  otherProp="some value"
  size={null}
  type="button"
  variant="primary"
/>
`;

exports[`Elements with analytics AnalyticsHDButtonDashed should render and match the snapshot 1`] = `
<HDButtonDashed
  className={null}
  disabled={false}
  icon={false}
  id="default-id"
  label={null}
  onClick={[Function]}
  onKeyDown={[Function]}
  otherProp="some value"
  tabIndex={0}
  theme="dark"
/>
`;

exports[`Elements with analytics AnalyticsHDCheckbox should render and match the snapshot 1`] = `
<HDCheckbox
  className="some-class-name"
  data="String"
  doReset={false}
  id="default-id"
  label="Some Label"
  name="toggle-button-name"
  onChange={[Function]}
  path="path.to.toggle-button"
  text="Sample text"
  type="checkbox"
  value="Some value"
/>
`;

exports[`Elements with analytics AnalyticsHDCheckboxButtonList should render and match the snapshot 1`] = `
<HDCheckboxButtonList
  className="Something"
  colProps={
    Object {
      "lg": 10,
      "xl": 8,
      "xs": 12,
    }
  }
  id="default-id"
  items={Array []}
  onChange={[Function]}
  selectedItems={Array []}
/>
`;

exports[`Elements with analytics AnalyticsHDCompletedCardInfo should render and match the snapshot 1`] = `
<HDCompletedCardInfo
  additionalText="Some additional text"
  className={null}
  deleteTabIndex={1}
  editTabIndex={0}
  id="default-id"
  onDelete={[Function]}
  onDeleteKeyDown={[Function]}
  onEdit={[Function]}
  onEditKeyDown={[Function]}
  text="Some sample text"
  variant="driver"
/>
`;

exports[`Elements with analytics AnalyticsHDDatePicker should render and match the snapshot 1`] = `
<HDDatePickerRefactor
  className=""
  customClassName=""
  hidePicker={false}
  id="default-id"
  information={null}
  initialDate={null}
  inputCols={Array []}
  inputStyle={null}
  label={null}
  maxDate={365}
  minDate={0}
  name="input-name"
  onBlur={[Function]}
  onChange={[Function]}
  onConfirm={[Function]}
  onSelect={[Function]}
  otherProp="some value"
  path="path.to.input"
  pickerPos="input-right"
  showFieldsNames={false}
  subLabel={null}
  theme="dark"
/>
`;

exports[`Elements with analytics AnalyticsHDDropdownList should render and match the snapshot 1`] = `
<HDDropdownList
  className=""
  customClassName=""
  data={null}
  defaultValue={null}
  displayValidationMessage={true}
  enableNative={false}
  id="default-id"
  isDisabled={false}
  isSearchable={true}
  label={null}
  name="dropdown-name"
  onBlur={[Function]}
  onChange={[Function]}
  options={
    Array [
      Object {
        "name": "name 1",
        "value": "value 1",
      },
      Object {
        "name": "name 2",
        "value": "value 2",
      },
    ]
  }
  otherProp="some value"
  path="path.to.dropdown"
  placeholder="Please select"
  ref={null}
  reset={false}
  selectSize={null}
  theme="white"
  titleTag="h5"
  value={null}
/>
`;

exports[`Elements with analytics AnalyticsHDLabel should render and match the snapshot 1`] = `
<HDLabelRefactor
  Tag="r"
  adjustImagePosition={false}
  className="label-with-icon align-items-start"
  href="https://www.hastingsdirect.com"
  iconPosition="l"
  id="default-id"
  onClick={[Function]}
  otherProp="some value"
  target="SomeTarget"
  text="Something"
/>
`;

exports[`Elements with analytics AnalyticsHDModal should render and match the snapshot 1`] = `
<ForwardRef
  id="default-id"
  onBeforeOpen={null}
  onBlur={null}
  onCancel={null}
  onChange={null}
  onClick={null}
  onConfirm={[Function]}
  onDelete={null}
  onEdit={null}
  onFocus={null}
  onSelect={null}
  otherProp="some value"
  show={true}
  webAnalyticsEvent={
    Object {
      "analytics_data": "top secret",
    }
  }
  webAnalyticsView={null}
>
  <div>
    Some children
  </div>
</ForwardRef>
`;

exports[`Elements with analytics AnalyticsHDOverlayPopup should render and match the snapshot 1`] = `
<ForwardRef
  id="default-id"
  onBeforeOpen={null}
  onBlur={null}
  onCancel={null}
  onChange={null}
  onClick={null}
  onConfirm={null}
  onDelete={null}
  onEdit={null}
  onFocus={null}
  onSelect={null}
  otherProp="some value"
  webAnalyticsEvent={
    Object {
      "analytics_data": "top secret",
    }
  }
  webAnalyticsView={null}
>
  <div>
    Some children
  </div>
</ForwardRef>
`;

exports[`Elements with analytics AnalyticsHDQuoteDownload should render and match the snapshot 1`] = `
<HDQuoteDownloadRefactor
  className=""
  id="default-id"
  linkText="Link"
  onClick={[Function]}
  otherProp="some value"
  showIcon={false}
/>
`;

exports[`Elements with analytics AnalyticsHDRadioButtonList should render and match the snapshot 1`] = `
<HDRadioButtonList
  className="Something"
  id="default-id"
  items={Array []}
  onChange={[Function]}
  value={
    Object {
      "id": "SomeID",
      "label": "Some Label",
      "value": "Required Value",
    }
  }
/>
`;

exports[`Elements with analytics AnalyticsHDSwitch should render and match the snapshot 1`] = `
<HDSwitch
  id="default-id"
  name="name"
  onChange={[Function]}
  otherProp="some value"
  path="path"
  theme="dark"
  value="name 1"
  values={
    Array [
      Object {
        "name": "name 1",
        "value": "value 1",
      },
      Object {
        "name": "name 2",
        "value": "value 2",
      },
    ]
  }
/>
`;

exports[`Elements with analytics AnalyticsHDTable should render and match the snapshot 1`] = `
<HDTable
  className=""
  data={Array []}
  defaultIndex={-1}
  headerValues={
    Array [
      Object {
        "headerLabelValue": "headerLabelValue",
        "image": <div>
          img
        </div>,
        "secondaryLabel": "secondaryLabel",
        "stickyHeaderText": "stickyHeaderText",
        "value": "value",
      },
    ]
  }
  hideBubbleOnHeaderSticky={false}
  id="default-id"
  moreDetailsLabel={null}
  moreDetailsPopups={Array []}
  name="name"
  onSelect={[Function]}
  onlineProduct={false}
  otherProp="some value"
  selectedHeaderValue={null}
  topHeader={null}
/>
`;

exports[`Elements with analytics AnalyticsHDTextInput should render and match the snapshot 1`] = `
<ForwardRef
  allowLeadingZero={false}
  appendContent={null}
  appendLabel={null}
  className=""
  customClassName=""
  data={null}
  disabled={false}
  icon={null}
  id="default-id"
  inputMode={null}
  isInvalidCustom={null}
  mask={null}
  maskChar="_"
  maxLength={null}
  name="input-name"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyPress={[Function]}
  otherProp="some value"
  path="path.to.input"
  placeholder={null}
  preText={null}
  reset={false}
  size={null}
  svgIcon={null}
  thousandSeprator={false}
  tickIcon={false}
  type={null}
  value=""
/>
`;

exports[`Elements with analytics AnalyticsHDToggleButtonGroup should render and match the snapshot 1`] = `
<HDToggleButtonGroup
  availableValues={
    Array [
      Object {
        "name": "name 1",
        "value": "value 1",
      },
      Object {
        "name": "name 2",
        "value": "value 2",
      },
    ]
  }
  btnClassName=""
  btnGroupClassName=""
  className=""
  customClassName=""
  data={null}
  disabled={false}
  doReset={false}
  id="default-id"
  label={null}
  name="toggle-button-name"
  onChange={[Function]}
  otherProp="some value"
  path="path.to.toggle-button"
  type="radio"
  value=""
/>
`;
